server/main-api/src/search/search_executor/lexer.rs

```diff
@@ -0,0 +1,167 @@
+use logos::{Lexer, Logos};
+
+/// An irregular split is defined as at least a letter and 1-4 numbers
+/// Treating words like MW1801 differently has improvements in relevancy for room-level searches
+fn irregular_split(lex: &mut Lexer<Token>) -> (String, String) {
+    let slice = lex.slice();
+    let mut split = slice.len();
+    for (i, c) in slice.char_indices().rev() {
+        if c.is_ascii_digit() {
+            split = i;
+        } else {
+            break;
+        }
+    }
+    let (text, numbers) = slice.split_at(split);
+    (text.to_string(), numbers.to_string())
+}
+
+/// Removes the specified prefix and additional whitespace from the token
+/// e.g used to remove the "in:" and "@" prefixes from filters
+fn remove_prefix(lex: &mut Lexer<Token>, prefix: &'static str) -> String {
+    lex.slice()[prefix.len()..].trim().to_string()
+}
+
+/// Parses the query string into a list of tokens
+/// priority between tokens is set as follows
+/// Filters|QuotedText > SplittableText > Text > skip
+#[derive(Logos, Debug, PartialEq, Eq, Clone)]
+#[logos(skip r"[ \t\n\f]+")]
+pub enum Token {
+    #[regex("\"[^\"]+\"", | lex | lex.slice()[1..lex.slice().len() - 1].to_string(), priority = 3)]
+    #[regex(r"[^ \t\n\f]+", | lex | lex.slice().to_string())]
+    Text(String),
+
+    #[regex("[a-zA-Z]+[0-9]{1,4}", irregular_split, priority = 2)]
+    SplittableText((String, String)),
+
+    #[regex("in: ?[a-zA-Z0-9]+", | lex | remove_prefix(lex, "in:"), priority = 3)]
+    #[regex("@ ?[a-zA-Z0-9]+", | lex | remove_prefix(lex, "@"), priority = 3)]
```

Comment:
```suggestion
    #[regex("in: ?[a-zA-Z0-9-]+", | lex | remove_prefix(lex, "in:"), priority = 3)]
    #[regex("@ ?[a-zA-Z0-9-]+", | lex | remove_prefix(lex, "@"), priority = 3)]
```
Some IDs have a `-` in them. Is there a reason you added the ` ?`? I don't think it's a problem (except in edge cases when you are about to prepend a filter to a query), but I also wouldn't expect it

Reply:
my assumption is that if you are writing `in: garching` a space can slip in, that is why.
Thinking about it: Lets include `.` here to 😅 
```suggestion
    #[regex("in: ?[a-zA-Z0-9-\.]+", | lex | remove_prefix(lex, "in:"), priority = 3)]
    #[regex("@ ?[a-zA-Z0-9-\.]+", | lex | remove_prefix(lex, "@"), priority = 3)]
```